File: src/main/java/com/github/kpavlov/jreactive8583/IsoMessageListener.java
package com.github.kpavlov.jreactive8583;

import com.solab.iso8583.IsoMessage;
import io.netty.channel.ChannelHandlerContext;

public interface IsoMessageListener<T extends IsoMessage> {

    /**
     * Returns <code>true</code> if given message can be handled by {@link #onMessage(ChannelHandlerContext, IsoMessage)}.
     *
     * @param isoMessage ISO message to check. Not null.
     * @return true if message should be handled.
     */
    boolean applies(T isoMessage);

    /**
     * Handles the message. If message should not be processed by any other handlers then this method should return <code>false</code>.
     *
     * @param ctx        current {@link ChannelHandlerContext}
     * @param isoMessage received isoMessage. Not null.
     * @return true if message should be handled by subsequent message listeners
     */
    boolean onMessage(ChannelHandlerContext ctx, T isoMessage);
}

File: src/main/kotlin/com/github/kpavlov/jreactive8583/IsoMessageListener.kt
@file:JvmName("IsoMessageListener")

package com.github.kpavlov.jreactive8583

import com.solab.iso8583.IsoMessage
import io.netty.channel.ChannelHandlerContext

public interface IsoMessageListener<T : IsoMessage> {
    /**
     * Returns `true` if given message can be handled
     * by [.onMessage].
     *
     * @param isoMessage ISO message to check. Not null.
     * @return true if message should be handled.
     */
    public fun applies(isoMessage: T): Boolean

    /**
     * Handles the message. If message should not be processed by any other handlers
     * then this method should return `false`.
     *
     * @param ctx        current [ChannelHandlerContext]
     * @param isoMessage received isoMessage. Not null.
     * @return true if message should be handled by subsequent message listeners
     */
    public fun onMessage(ctx: ChannelHandlerContext, isoMessage: T): Boolean
}
